### PR TITLE
UI elements for create cluster through ACM

### PR DESCRIPTION
**Pull request by shylesh**
Signed-off-by: Shylesh Kumar Mohan <shmohan@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/ui/views.py [hunk 1]
```diff
@@ -453,6 +454,80 @@
         By.XPATH,
     ),
     "cluster-set-selection": ("//a[normalize-space()='{}']", By.XPATH),
+    "cc_create_cluster": ("createCluster", By.ID),
+    "cc_provider_vmware_vsphere": ("//*[@id='vmware-vsphere']", By.XPATH),
+    "cc_cluster_name": ("input[placeholder='Enter cluster name']", By.CSS_SELECTOR),
```
Comment (am-agrawa) [inline ocs_ci/ocs/ui/views.py:459]: How will this xpath utilize the cluster name?

File: ocs_ci/ocs/ui/views.py [hunk 2]
```diff
@@ -453,6 +454,80 @@
...
+    "cc_base_dns_domain": (
+        "input[placeholder='Enter base DNS domain'",
```
Comment (am-agrawa) [inline ocs_ci/ocs/ui/views.py:461]: Same

File: ocs_ci/ocs/ui/views.py [hunk 3]
```diff
@@ -453,6 +454,80 @@
...
+        By.CSS_SELECTOR,
+    ),
+    "cc_openshift_release_image": (
+        "input[placeholder='Select or enter a release image']",
```
Comment (am-agrawa) [inline ocs_ci/ocs/ui/views.py:465]: Same

File: ocs_ci/ocs/ui/views.py [hunk 4]
```diff
@@ -453,6 +454,80 @@
...
+        By.CSS_SELECTOR,
+    ),
+    "cc_vsphere_network_name": (
+        "input[placeholder='vSphere network name']",
```
Comment (am-agrawa) [inline ocs_ci/ocs/ui/views.py:469]: Please re-check this CSS Selector

File: ocs_ci/ocs/ui/views.py [hunk 5]
```diff
@@ -453,6 +454,80 @@
...
+        By.CSS_SELECTOR,
+    ),
+    "cc_api_vip": ("input[placeholder='Enter api vip']", By.CSS_SELECTOR),
```
Comment (am-agrawa) [inline ocs_ci/ocs/ui/views.py:472]: Please re-check this CSS Selector

File: ocs_ci/ocs/ui/views.py [hunk 6]
```diff
@@ -453,6 +454,80 @@
...
+    "cc_ingress_vip": ("input[placeholder='Enter ingress VIP'", By.CSS_SELECTOR),
```
Comment (am-agrawa) [inline ocs_ci/ocs/ui/views.py:473]: Please re-check this CSS Selector

File: ocs_ci/ocs/ui/views.py [hunk 7]
```diff
@@ -453,6 +454,80 @@
...
+    "cc_next_page_button": ("//button[normalize-space()='Next']", By.XPATH),
+    "cc_create_button": ("//button[normalize-space()='Create']", By.XPATH),
+    "cc_cluster_details": ("//div[contains(text(),'Cluster details')]", By.XPATH),
+    "cc_node_pools": ("//div[contains(text(),'Node pools')]", By.XPATH),
+    "cc_networks": ("//div[contains(text(),'Networks')]", By.XPATH),
+    "cc_proxy": ("//div[contains(text(),'Proxy')]", By.XPATH),
+    "cc_review": ("//button[normalize-space()='Review']", By.XPATH),
+    "cc_infrastructure_provider_creds": (
+        "input[placeholder='Select a credential']",
```
Comment (am-agrawa) [inline ocs_ci/ocs/ui/views.py:482]: How will credentials be passed into the XPath?

**Review by am-agrawa — COMMENTED** [review]
Please re-visit

File: ocs_ci/ocs/ui/views.py (hunk 7 above)
Comment (shylesh) [reply ocs_ci/ocs/ui/views.py:482]: So, the Idea is, by default its a dropdown, I would have created a vsphere cred secret from CLI, then click on this dropdown and select that vsphere cred (whose xpath needs to be dynamically generated at run time).

File: ocs_ci/ocs/ui/views.py (hunk 1 above)
Comment (shylesh) [reply ocs_ci/ocs/ui/views.py]: changed to xpath.

File: ocs_ci/ocs/ui/views.py (hunk 2 above)
Comment (shylesh) [reply ocs_ci/ocs/ui/views.py]: changed to xpath.

File: ocs_ci/ocs/ui/views.py (hunk 3 above)
Comment (shylesh) [reply ocs_ci/ocs/ui/views.py]: changed to xpath.

File: ocs_ci/ocs/ui/views.py (hunk 4 above)
Comment (shylesh) [reply ocs_ci/ocs/ui/views.py]: changed to xpath.

File: ocs_ci/ocs/ui/views.py (hunk 5 above)
Comment (shylesh) [reply ocs_ci/ocs/ui/views.py]: changed to xpath.

File: ocs_ci/ocs/ui/views.py (hunk 6 above)
Comment (shylesh) [reply ocs_ci/ocs/ui/views.py]: changed to xpath.

File: ocs_ci/ocs/ui/views.py (hunk 7 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/views.py:473]: exactly, so you should use {} in the place of `Select a credential` and then pass the cred at run time using the format_locator function we have in helpers_ui file

**Review by petr-balogh — DISMISSED** [review]

File: ocs_ci/ocs/ui/views.py (hunk 7 above)
Comment (shylesh) [reply ocs_ci/ocs/ui/views.py:473]: done

**Review by am-agrawa — APPROVED** [review]
LGTM, will need to check at the time of actual code implementation.

**Pull request merged**